src/App/Document.cpp

```diff
@@ -192,17 +198,24 @@ struct DocumentP
     std::multimap<const App::DocumentObject*,
         std::unique_ptr<App::DocumentObjectExecReturn> > _RecomputeLog;
 
+    StringHasherRef Hasher;
+
     // restored files
     std::set<std::string> files;
 
     DocumentP() {
+#ifndef FC_DEBUG
         static std::random_device _RD;
         static std::mt19937 _RGEN(_RD());
         static std::uniform_int_distribution<> _RDIST(0,5000);
         // Set some random offset to reduce likelihood of ID collision when
         // copying shape from other document. It is probably better to randomize
         // on each object ID.
         lastObjectId = _RDIST(_RGEN);
+#else
+        lastObjectId = 10; 

```

Comment:
Did you mean to include this in your PR? It looks like testing code.

Reply:
The reason of using a randomized starting ID is stated in the comment. But when building debug version it will be better to have reproducible results. The object ID is encoded into the topo naming.

Reply:
OK -- I was asking out of self-preservation, I nearly always run FreeCAD in debug mode so if something breaks I have a chance of figuring out what happened. I want to make sure my copy of FreeCAD works 😄 .